src/clarity-angular/wizard/providers/wizard-navigation.ts

```diff
@@ -164,50 +125,100 @@ export class WizardNavigationService implements OnDestroy {
     // This is a public function that can be used to programmatically advance
     // the user to the next page.
     public next(): void {
-        let currentPage = this.currentPage;
-        let nextPage: WizardPage;
-
         if (this.currentPageIsLast) {
             this.finish();
         }
 
-        if (!currentPage.readyToComplete) {
-            return;
-        }
+        this.forceNext();
 
-        if (!currentPage.preventDefault) {
-            // when preventDefault is enacted, this is called manually
-            // after an onCommit. put check in here to avoid duplicate
-            // onCommit calls
-            this.pageCollection.commitPage(currentPage);
+        if (!this.wizardHasAltNext) {
+            this._movedToNextPage.next(true);
         }
-        nextPage = this.pageCollection.getNextPage(currentPage);
+        // SPECME
+    }
+
+    public forceNext(): void {
+        let currentPage: WizardPage = this.currentPage;
+        let nextPage: WizardPage = this.pageCollection.getNextPage(currentPage);
 
         // catch errant null or undefineds that creep in
-        if (nextPage) {
-            this.setCurrentPage(nextPage);
-            this._movedToNextPage.next(true);
-        } else {
+        if (!nextPage) {
             throw new Error("The wizard has no next page to go to.");
         }
+
+        if (!currentPage.completed) {
+            // this is a state that alt next flows can get themselves in...
+            this.pageCollection.commitPage(currentPage);
+        }
+        this.setCurrentPage(nextPage);
+        // SPECME
     }
 
-    public finish(): void {
-        let currentPage = this.currentPage;
+    public checkAndCommitCurrentPage(buttonType: string): void {
```

Comment:
more of a stylistic question. i usually declare variables at the top and then do early returns.